tests/routers/test_remote.py

```
@@ -584,114 +782,97 @@ def test_pull_file_success(
-    # The global mock_verify_remote_signature returns a MagicMock connection,
-    # we need to ensure its remote_transfer_mode and effective_bidirectional are updated
-    mock_remote_connection.remote_transfer_mode = TransferMode.PUSH
-    mock_remote_connection.effective_bidirectional = True
-    
-    # Ensure our global mock_verify_remote_signature returns this mock connection
-    # Note: In a real scenario, mock_verify_remote_signature would return the specific conn
-    # related to the signature. For this test, we are just making sure the patching works.
-    with patch("app.routers.api.remote.verify_remote_signature", return_value=mock_remote_connection):
+
+    # Override verify_remote_signature to return this specific connection
+    async def _mock_specific_connection():
```

Ruff (0.15.1)

[warning] Missing return type annotation for private function `_mock_specific_connection`

(ANN202)